src/distances.jl

```diff
@@ -24,64 +84,47 @@ julia> Ellipsoidal([1.0,0.5], [π/2])
 julia> Ellipsoidal([1.0,0.5,0.5], [π/2,0.0,0.0])
 ```
 """
+
 struct Ellipsoidal{N,T} <: Metric
   dist::Mahalanobis{T}
 
-  function Ellipsoidal{N,T}(semiaxes, angles) where {N,T}
+  function Ellipsoidal{N,T}(semiaxes, angles; convention=:TaitBryanExtr) where {N,T}
     @assert length(semiaxes) == N "number of semiaxes must match spatial dimension"
     @assert all(semiaxes .> zero(T)) "semiaxes must be positive"
     @assert N ∈ [2,3] "dimension must be either 2 or 3"
 
-    # scaling matrix
-    Λ = Diagonal(one(T)./semiaxes.^2)
-
-    # rotation matrix
-    if N == 2
-      θ = angles[1]
-
-      cosθ = cos(θ)
-      sinθ = sin(θ)
+    rule = rules[convention]
 
-      P = [cosθ -sinθ
-           sinθ  cosθ]
+    # invert x and y if necessary
+    if rule.main == :y
+       semiaxes[1], semiaxes[2] = semiaxes[2], semiaxes[1]
     end
-    if N == 3
-      θxy, θyz, θzx = angles
 
-      cosxy = cos(θxy)
-      sinxy = sin(θxy)
-      cosyz = cos(θyz)
-      sinyz = sin(θyz)
-      coszx = cos(θzx)
-      sinzx = sin(θzx)
-
-      _1 = one(T)
-      _0 = zero(T)
-
-      Rxy = [cosxy -sinxy _0
-             sinxy  cosxy _0
-             _0     _0 _1]
-
-      Ryz = [_1    _0     _0
-             _0 cosyz -sinyz
-             _0 sinyz  cosyz]
+    # scaling matrix
+    Λ = Diagonal(one(T)./semiaxes.^2)
+    #Λ = SMatrix{N,N}(Diagonal(one(T)./semiaxes.^2))
```

Comment:
I think we can just `SMatrix(Diagonal(...))`, we don't need the `{N,N}` there.

Reply:
Nevermind, we actually need the sizes explicitly with matrices.

Reply:
I think this one can be `SDiagonal(ntuple(i->one(T)/semiaxes[i]^2, N))`

Reply:
Done